src/Microsoft.AspNetCore/WebHost.cs

```diff
@@ -0,0 +1,195 @@
+        ///     set the <see cref="IHostingEnvironment.ContentRootPath"/> to the result of <see cref="Directory.GetCurrentDirectory()"/>,
+        ///     load <see cref="IConfiguration"/> from 'appsettings.json' and 'appsettings.[<see cref="IHostingEnvironment.EnvironmentName"/>].json',
+        ///     load <see cref="IConfiguration"/> from User Secrets when <see cref="IHostingEnvironment.EnvironmentName"/> is 'Development' using the entry assembly,
+        ///     load <see cref="IConfiguration"/> from environment variables,
+        ///     load <see cref="IConfiguration"/> from supplied command line args,
+        ///     configures the <see cref="ILoggerFactory"/> to log to the console,
+        ///     enables IIS integration,
+        ///     and adds the developer exception page when <see cref="IHostingEnvironment.EnvironmentName"/> is 'Development'
+        /// </remarks>
+        /// <param name="args">The command line args.</param>
+        /// <returns>The initialized <see cref="IWebHostBuilder"/>.</returns>
+        public static IWebHostBuilder CreateDefaultBuilder(string[] args)
+        {
+            var builder = new WebHostBuilder()
+                .UseKestrel()
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .ConfigureConfiguration((hostingContext, config) =>
+                {
+                    var env = hostingContext.HostingEnvironment;
+
+                    config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
+
+                    if (env.IsDevelopment())
+                    {
+                        var appAssembly = GetApplicationAssembly();
+                        if (appAssembly != null)
+                        {
+                            config.AddUserSecrets(appAssembly);
+                        }
+                    }
+
+                    config.AddEnvironmentVariables();
+
+                    if (args != null)
+                    {
+                        config.AddCommandLine(args);
+                    }
+                })
+                .ConfigureLogging(logging =>
+                {
+                    logging.AddConsole();
+                })
+                // TODO: Remove this when ANCM injects it by default
+                .UseIISIntegration()
+                .ConfigureServices(services =>
+                {
+                    services.AddSingleton<IStartupFilter, WebHostStartupFilter>();
+                });
+
+            return builder;
+        }
+
+        private static Assembly GetApplicationAssembly()
+        {
+            try
+            {
+                // Copied from https://github.com/aspnet/PlatformAbstractions/blob/dev/src/Microsoft.Extensions.PlatformAbstractions/ApplicationEnvironment.cs#L49
+                var getEntryAssemblyMethod =
```

Comment:
Should we cross-target to ns1.5 and only put an assembly in that lib, or the entire meta-package?